### PR TITLE
[exporter/opensearch] Add OpenSearchExporter to otelcontribcol

### PR DESCRIPTION
**Pull request by MaxKsyunz**
This PR completes addition of OpenSearch exporter.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
LGTM. Note that this PR does **not** make the open search exporter available in the Contrib distribution. To do this, you need to file a PR against the manifest available over at opentelemetry-collector-releases

File: .chloggen/add-opensearchexporter.yaml
Comment (dmitryax) [inline .chloggen/add-opensearchexporter.yaml:7]: Adding the exporter to the development build isn't a customer-facing change. This line should call out the promotion of the component to the Alpha stability level instead.

File: .chloggen/add-opensearchexporter.yaml
Comment (dmitryax) [inline .chloggen/add-opensearchexporter.yaml:3]:
```suggestion
change_type: enhancement

```

Given that it was already mentioned as a new component in 0.82.0

**Review by fatsheep9146 — APPROVED** [review]

**Comment by MaxKsyunz** [comment]
@dmitryax your comments have been addressed -- thanks for taking a look.

**Comment by MaxKsyunz** [comment]
@mx-psi is this good to merge?

**Comment by mx-psi** [comment]
@dmitryax PTAL!

**Comment by MaxKsyunz** [comment]
ref https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26560

**Comment by mx-psi** [comment]
ping @dmitryax

**Comment by MaxKsyunz** [comment]
There is [WIP](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/26475) to send logs to OpenSearch.

**Comment by mx-psi** [comment]
@MaxKsyunz can you push an empty commit to retrigger CI? Seems like something went wrong on Github's side on the last commit

**Comment by MaxKsyunz** [comment]
@mx-psi do you know what's keeping this PR from getting `ready to merge` label?

All the feedback has been addressed, there's end user interest and there are couple other PRs that are blocked on this.

**Review by YANG-DB — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Comment by mx-psi** [comment]
>  do you know what's keeping this PR from getting ready to merge label? Once I get thumbs of from someone, I'll address merge conflicts.

I was waiting on @dmitryax to give their approval. Since we have an approval from someone else we can merge once you solve the merge conflicts

**Comment by mx-psi** [comment]
@MaxKsyunz Gentle reminder that I will merge this once merge conflicts are addressed :)

**Comment by MaxKsyunz** [comment]
> @MaxKsyunz Gentle reminder that I will merge this once merge conflicts are addressed :)

@mx-psi my apologies -- I was away from GitHub. Thank you for keeping this PR alive. :)

I'll push an update soon.

**Comment by sumobrian** [comment]
Are the unresolved test failures preventing this branch from being merged? The branch conflicts were addressed last week, but new conflicts have since emerged.

**Comment by mx-psi** [comment]
> The branch conflicts were addressed last week, but new conflicts have since emerged.

There are still branch conflicts, we can't merge until these are solved and tests pass:

image

**Comment by sumobrian** [comment]
> > The branch conflicts were addressed last week, but new conflicts have since emerged.
> 
> There are still branch conflicts, we can't merge until these are solved and tests pass:
> 
> image

Copy. For a brief period there were no conflicts BUT all tests were not passing.

**Comment by MaxKsyunz** [comment]
@mx-psi all checks are green.

**Pull request merged**
